### PR TITLE
Adds keras metrics

### PR DESCRIPTION
**Pull request by joaogui1**
Hi, this is a partial solution to #8 by using a Metric object's name, this way the library is now compatible with all keras metrics. If someone wants to point me toward a more complete solution I would be more than happy to implement it!

### CODE REVIEW

**Comment by razorx89** [comment]
This fix does not work for all tf.keras.metrics.Metric derived (custom) metrics. Even changing the name of a tf.keras.metric.* metric breaks with a ValueError, since the name cannot be found in the tf.keras.metrics module.

**Comment by joaogui1** [comment]
Sorry, I meant (though indeed the way I wrote it wasn't clear) that it was now compatible with passing an unmodified Keras metric object. Custom metrics will indeed still not work.

**Comment by joaogui1** [comment]
To be clearer this is a quick solution (around 2 lines of code) that solves a small part of the problem.
I tested the code a lot and didn't manage to understand the exact origin of the bug. If someone wants to talk to me so I can explain what I did or point me towards a way to implement a solution I would be happy to help!

**Comment by csferng** [comment]
@joaogui1 Thanks for the suggested fix.

If a `Metric` object is given (no matter if it's in the Keras library), the same object should be used to calculate metric values, so that any downstream use or side effect of the `Metric` object can still work. Creating a new `Metric` object of the same name doesn't meet this need.

I have a rough idea of how to fix this, although haven't sorted out some details yet. Since the actual fix will likely involve multiple rounds of trial and error, I will do this in a separate commit.

**Comment by joaogui1** [comment]
Okss, closing this PR
